publicroomsapi/publicroomsapi.go

```diff
@@ -35,7 +35,13 @@ func SetupPublicRoomsAPIComponent(
 	fedClient *gomatrixserverlib.FederationClient,
 	extRoomsProvider types.ExternalPublicRoomsProvider,
 ) {
-	publicRoomsDB, err := storage.NewPublicRoomsServerDatabase(string(base.Cfg.Database.PublicRoomsAPI))
+	var err error
+	var publicRoomsDB storage.Database
+	if base.LibP2P != nil {
+		publicRoomsDB, err = storage.NewPublicRoomsServerDatabaseWithPubSub(string(base.Cfg.Database.PublicRoomsAPI), base.LibP2PPubsub)
+	} else {
+		publicRoomsDB, err = storage.NewPublicRoomsServerDatabase(string(base.Cfg.Database.PublicRoomsAPI))
```

Comment:
I'd prefer it if you changed the function signature of `SetupPublicRoomsAPIComponent` to accept a `publicRoomDB` then your `/cmd` can do this without needing to switch on `base.LibP2P`. This code originally should've injected all of its dependencies but I guess it didn't for brevity.

Reply:
Done in 89943bf.